Modules/Microsoft365DSC/DSCResources/MSFT_AADNamedLocationPolicy/MSFT_AADNamedLocationPolicy.psm1

```diff
@@ -62,7 +62,17 @@ function Get-TargetResource
     Write-Verbose -Message "Getting configuration of AAD Named Location"
 
     $ConnectionMode = New-M365DSCConnection -Workload 'MicrosoftGraph' `
-        -InboundParameters $PSBoundParameters
+        -InboundParameters $PSBoundParameters `
+        -ProfileName 'v1.0'
+    $context=Get-MgContext
+    if($null -eq $context)
+    {
+        $ConnectionMode = New-M365DSCConnection -Workload 'MicrosoftGraph' `
```

Comment:
Hi @NikCharlebois 
I experience some inconsistence when using New-M365DSCConnection -Workload 'MicrosoftGraph' resulting in my connection failing depending if the script is run the first time of a pssession or after subsequent runs. For example, if I run the command on a new pssession with -ProfileName 'v1.0' it will fail but -ProfileName 'beta' will succeed. 
And if I run the script again in the same session, -ProfileName 'v1.0' will succeed and -ProfileName 'beta' will fail.

So to make sure the connection works every time I use Get-MgContext to identify if the connection was successful with New-M365DSCConnection -Workload 'MicrosoftGraph' `
            -InboundParameters $PSBoundParameters `
            -ProfileName 'v1.0' 
and if not I use the beta profile.

This is the only way I was able to get a consistent connection to the Graph API in my environment - I can remove that logic from the resource if you prefer
Thanks
Will

Reply:
Thanks for the clarifications. Let's approve the change for now, but I am curious to dig deeper into the source of the issue. Are you also encountering the same behavior in other resources as well?